pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -115,11 +115,12 @@ public SocketFactory getSocketFactory() {
    * @throws IOException if something wrong happens
    */
   public boolean hasMessagePending() throws IOException {
+    long now = System.nanoTime();
```

Comment:
@bokken @vlsi thoughts on this ? This seems to stop the problem, but I think it is masking the problem. Either way I think this change is correct as it correctly assesses idle time now

Reply:
I think this makes it less likely for peek to get called, which might just be impacting the buffering.

Reply:
Well the other way the code was written I suspect it was calling peek quite often.

Reply:
Except (as commented above) I think there was a conversion mistake to nanos which likely cause peek to be called even more often. That could explain why it decreased occurrence of issue.

Reply:
possibly, except @omgrr can replicate it easily either way. 
I agree this is a mistake in conversion